dashboard/app/models/ability.rb

```diff
@@ -485,17 +485,13 @@ def initialize(user)
         user.verified_instructor? || user.sections_as_student.any? {|s| s.assigned_csa? && s.teacher&.verified_instructor?}
       end
 
-      if user.has_pilot_experiment?(GENAI_PILOT) ||
-          (!user.teachers.empty? &&
-          user.teachers.any? {|teacher| teacher.has_pilot_experiment?(GENAI_PILOT)})
-        can :chat_completion, :aichat
-        can :log_chat_event, :aichat
-        can :start_chat_completion, :aichat
-        can :chat_request, :aichat
-      end
-      # Only teachers can view student chat history.
-      if user.has_pilot_experiment?(GENAI_PILOT)
-        can :student_chat_history, :aichat
+      can [:chat_completion, :log_chat_event, :start_chat_completion, :chat_request], :aichat do
```

Comment:
Some (seemingly unrelated) tests that check the # of database queries that are performed in a given controller action were failing, and this turned out to be the culprit.

TIL that logic that is more complex should be put inside a block like this, rather than at the top level when a new Ability object is instantiated. As previously written, we were running this check any time we're checking user permissions for anything -- now, we'll just run it when we're specifically checking for access to AI Chat.

Reply:
Oh interesting! Thanks for the explanation.

Reply:
Can we remove the constant `GENAI_PILOT` on line 5?

Reply:
Good call, done!